pkg/zsio/ndjson/writer.go

```diff
@@ -10,22 +10,17 @@ import (
 // NDJSON implements a Formatter for ndjson
 type NDJSON struct {
 	zson.Writer
+	encoder *json.Encoder
 }
 
 func NewWriter(w io.WriteCloser) *NDJSON {
-	return &NDJSON{Writer: zson.Writer{w}}
+	writer := &NDJSON{
+		Writer: zson.Writer{w},
+	}
+	writer.encoder = json.NewEncoder(writer.Writer)
+	return writer
```

Comment:
Nit: I think you can simplify this to
```go
	return &NDJSON{
		Writer: zson.Writer{w},
		encoder: json.NewEncoder(w),
	}
```

Reply:
I would call this Writer instead of NDJSON (my bad when I created the stub), i.e., ndjson.Writer

Reply:
Oh yeah, duh